tests/app/delivery/test_send_to_providers.py

```diff
@@ -888,6 +888,50 @@ def test_notification_document_with_pdf_attachment(
         assert call(statsd_key, notification.sent_at, notification.created_at) in statsd_calls
         assert call(statsd_key) in statsd_mock.incr.call_args_list
 
+@pytest.mark.parametrize(
+    "filename_attribute_present, filename, expected_filename",
+    [
+        (True, None, None),
+        (True, "custom_filename.pdf", "custom_filename.pdf"),
+    ],
+)
+def test_notification_document_with_illegal_url_attachment(
+    mocker,
+    notify_db,
+    notify_db_session,
+    filename_attribute_present,
+    filename,
+    expected_filename,
+):
+    template = sample_email_template(notify_db, notify_db_session, content="Here is your ((file))")
+    personalisation = {
+        "file": document_download_response(
+            {
+                "direct_file_url": "file://foo.bar/direct_file_url",
+                "url": "http://foo.bar/url",
+                "mime_type": "application/pdf",
+                "mlwr_sid": "false",
+            }
+        )
+    }
+    if filename_attribute_present:
+        personalisation["file"]["document"]["filename"] = filename
+        personalisation["file"]["document"]["sending_method"] = "attach"
+    else:
+        personalisation["file"]["document"]["sending_method"] = "link"
+
+    db_notification = create_notification(template=template, personalisation=personalisation)
+
+    # See https://stackoverflow.com/a/34929900
```

Comment:
I would provide a summary of the issue here along with the SO link.

Reply:
Copied this down from test_notification_document_with_pdf_attachment. Not sure what the underlying issue, just that i seem to need it to get the test to work :)

Reply:
Damn...